llama-index-core/llama_index/core/indices/base.py

```diff
@@ -48,7 +48,7 @@ def __init__(
         index_struct: Optional[IS] = None,
         storage_context: Optional[StorageContext] = None,
         callback_manager: Optional[CallbackManager] = None,
-        transformations: Optional[List[TransformComponent]] = None,
+        transformations: Optional[Sequence[TransformComponent]] = None,
```

Comment:
`BaseIndex` does not require that `transformations` specifically be a list. Existing subclasses that assume that it is a list should remain unaffected (in terms of type safety) as long as they specify `transformations` as a list in the subclass's initializer.